src/brsTypes/components/RoSGNode.ts

```diff
@@ -1320,6 +1321,13 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
 export const mGlobal = new RoSGNode([]);
 
 export function createNodeByType(interpreter: Interpreter, type: BrsString): RoSGNode | BrsInvalid {
+    // If the requested component has been mocked, construct and return the mock instead
+    let maybeMock = interpreter.environment.getMockObject(type.value.toLowerCase());
+    if (maybeMock instanceof RoAssociativeArray) {
+        let mock: typeof MockNodeModule = require("../../extensions/MockNode");
```

Comment:
Copy-pasted (ish) from https://www.typescriptlang.org/docs/handbook/modules.html#optional-module-loading-and-other-advanced-loading-scenarios